src/replit_river/v2/session.py

```diff
@@ -397,16 +411,31 @@ async def _enqueue_message(
         self._process_messages.set()
 
     async def close(
-        self, reason: Exception | None = None, current_state: SessionState | None = None
+        self,
+        reason: Exception | None = None,
+        current_state: SessionState | None = None,
+        _wait_for_closed: bool = True,
```

Comment:
It seems like we will always wait for closed if the `_closing_waiter` isn't set yet, even if this is set to `False`?

Reply:
The boolean here was admittedly a hack to keep moving, let me see if I can pull it out